### PR TITLE
Clean unwrap again

### PR DESCRIPTION
**Pull request by darnuria**
An other pull request who try to remove `unwraps calls`.

If any commit is better to be dropped tell me! On to roll-up if needed.
I chose to remove some unwrap who are "not possible" rustc can know it and lift the `?` branch (because where is a size check) and it's future proof and fool prof.

May check cargo asm of the function to check my statement in a comment later.

### CODE REVIEW

File: crates/server/src/store/sqlite/persistence.rs [hunk 1]
```diff
@@ -42,13 +42,9 @@ impl Journal {
...
-        match result {
-            Ok(mut journal) => {
-                journal.schema_up().unwrap();
-                Ok(journal)
-            }
-            Err(err) => Err(err),
-        }
+        let mut journal = result?;
+        journal.schema_up()?;
+        Ok(journal)
```
Comment (darnuria) [inline crates/server/src/store/sqlite/persistence.rs:47]: As I expected assembly is the same checked with `cargo asm --rust trust_dns_server::store::sqlite::persistence::Journal::from_file` `?` got quite optimized away well.

**Review by djc — APPROVED** [review]
Good stuff!

**Comment by djc** [comment]
(I would merge this, but I'm not allowed to due to the clippy failures. @bluejekyll maybe consider making the branch protection a little less strict and/or giving me admin privileges so I can bypass?)

**Review by bluejekyll — APPROVED** [review]
this looks great to me.

**Comment by bluejekyll** [comment]
I'll merge, we can take care of the clippy failures separately.

**Comment by darnuria** [comment]
thanks! Had done a second PR about the clippy error no problem with closing it if it's not relevant.

**Pull request merged**